client/my-sites/plugins/marketplace/marketplace-plugin-details/index.tsx

```diff
@@ -73,11 +82,27 @@ function MarketplacePluginDetails( {
 		return replaceProductsInCart( [] );
 	};
 
+	const allBlockingMessages = getBlockingMessages( translate );
+	const holds = eligibilityDetails.eligibilityHolds || [];
+	const raisedBlockingMessages = holds
+		.filter( ( message: string ) => allBlockingMessages[ message ] )
+		.map( ( message: string ) => allBlockingMessages[ message ] );
+	const hardBlockSingleMessages = holds.filter(
```

Comment:
From the usage below, i.e. `hardBlockSingleMessages.length > 0`, all we need to know is if there is a message in `holds` array that is neither `TRANSFER_ALREADY_EXISTS` nor a key in `allBlockingMessages`. However, `filter()` will go through the entire array even if the 1st element has fulfilled the condition already; the produced filtered array is also not used any other way.

I'd suggest us to use `find()` or any other simpler function in this case. e.g.

```
const hasHardBlockSingleMessage = holds.find( ( message: string ) => message !== 'TRANSFER_ALREADY_EXISTS' || ! allBlockingMessages[ message ];
...
const hasHardBlock = `isAtomicSiteWithoutBusinessPlan( holds ) || hasHardBlockSingleMessage;
```

It's minor though, so please feel free to take it or not :)

Reply:
Hey that is a great tip @southp 🙇 
Also, I took it a step further and used [Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)